kodi_addon_checker/check_repo.py

```diff
@@ -24,21 +17,22 @@ def check_repo(repo_path, parameters):
 
     print("Toplevel folders " + str(toplevel_folders))
 
-    config = _read_config_for_version(repo_path)
-
     for addon_folder in toplevel_folders:
         if addon_folder[0] != '.':
+            repo_report.add(Record(INFORMATION, "Checking add-on %s" % addon_folder))
             addon_path = os.path.join(repo_path, addon_folder)
-            error_counter = check_addon.start(
-                error_counter, addon_path, config)
+            addon_report = check_addon.start(addon_path, config)
+            repo_report.add(addon_report)
+
+    ReportManager.report(repo_report)
 
-    if error_counter["problems"] > 0:
+    if repo_report.problem_count > 0:
         colorPrint("We found %s problems and %s warnings, please check the logfile." % (
```

Comment:
Is this correct? Shouldn't this also use the reporter and just create a record?

Reply:
Fixed